de.metas.adempiere.adempiere/base/src/main/java-legacy/org/adempiere/model/RelationTypeZoomProvider.java

```diff
@@ -84,8 +85,22 @@ private RelationTypeZoomProvider(final Builder builder)
 		internalName = builder.getInternalName();
 		adRelationTypeId = builder.getAD_RelationType_ID();
 
-		source = new ZoomProviderDestination(builder.getSource_Reference_ID(), builder.getSourceTableRefInfoOrNull(), builder.getSourceRoleDisplayName());
-		target = new ZoomProviderDestination(builder.getTarget_Reference_ID(), builder.getTargetTableRefInfoOrNull(), builder.getTargetRoleDisplayName());
+		isReferenceTarget = builder.isReferenceTarget();
+
+		if (!isReferenceTarget)
+		{
+			source = new ZoomProviderDestination(
+					builder.getSource_Reference_ID(),
+					builder.getSourceTableRefInfoOrNull(),
+					builder.getSourceRoleDisplayName(),
+					false // builder.isReferenceTarget
+			);
+		}
+		else
+		{
+			source = null;
```

Comment:
A user of RelationTypeZoomProvider might now run into an NPE when calling `RelationTypeZoomProvider .getTableName()`.
Please avoid such a situation. You could e.g. introduce a dedicated ZoomProviderDestination for reference types.

Reply:
I refactored all this part. Please, review.